### PR TITLE
Teleport Machine ID `tbot`

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/roadmap/issues/2948

### What this PR does / why we need it
- Adds support for Teleport Machine ID Bot for short-lived certificate for Teleport Cluster access.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] tbot deployment works and generates the `identity-output` secret with short-lived certificate.
- [x] teleport-operator re-read the `identity-output` secret, as tbot rotates the short-lived certificate every 20 minutes.

### CODE REVIEW

File: internal/pkg/teleport/secret.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type SecretConfig struct {
 
 func GetConfigFromSecret(ctx context.Context, ctrlClient client.Client, namespace string) (*SecretConfig, error) {
 	secret := &corev1.Secret{}
```
Comment (kopiczko) [inline internal/pkg/teleport/secret.go:29]: Doesn't look like any information in the secret is secret now

File: internal/pkg/teleport/secret.go
Comment (kopiczko) [inline internal/pkg/teleport/secret.go:1]: Do I understand correctly secret refreshing is not implemented yet?

Comment (tuladhar) [reply internal/pkg/teleport/secret.go:1]: Yes

File: internal/pkg/teleport/secret.go (hunk 1 above)
Comment (tuladhar) [reply internal/pkg/teleport/secret.go:29]: Right, should go to config map.

File: internal/pkg/teleport/secret.go [hunk 2]
```diff
@@ -35,12 +36,19 @@ func GetConfigFromSecret(ctx context.Context, ctrlClient client.Client, namespac
 		return nil, microerror.Mask(err)
 	}
 
+	if err := ctrlClient.Get(ctx, types.NamespacedName{
+		Name:      key.TeleportBotSecretName,
```
Comment (anvddriesch) [inline internal/pkg/teleport/secret.go:40]: Do I understand correctly that `tbot` will create this identity secret which can then be picked up by the operator?

Comment (tuladhar) [reply internal/pkg/teleport/secret.go:40]: Yes, that's correct.

**Review by anvddriesch — APPROVED** [review]
lgtm

File: internal/controller/cluster_controller.go [hunk 3]
```diff
@@ -68,6 +71,32 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+	now := time.Now()
+	diff := now.Sub(r.Teleport.SecretConfig.LastRead)
+	seconds := diff.Seconds()
+	minutes := seconds / 60
+	hasher := sha512.New()
+	hasher.Write([]byte(r.Teleport.SecretConfig.IdentityFile))
+	sum := hasher.Sum(nil)
+	hashString := hex.EncodeToString(sum)
+
+	log.Info("Teleport identity", "last-read-minutes-ago", minutes, "hash", hashString)
```
Comment (tuladhar) [inline internal/controller/cluster_controller.go:83]: We can do this better, and separate it out.

**Review by ssyno — APPROVED** [review]

**Pull request merged**